matlab/CMakeLists.txt

```diff
@@ -76,108 +96,96 @@ function(build_arrow)
   file(MAKE_DIRECTORY "${ARROW_INCLUDE_DIR}")
   add_library(${ARROW_LIBRARY_TARGET} SHARED IMPORTED)
   set_target_properties(${ARROW_LIBRARY_TARGET}
-                        PROPERTIES ${ARROW_IMPORTED_TYPE} ${ARROW_SHARED_LIB}
-                                   INTERFACE_INCLUDE_DIRECTORIES ${ARROW_INCLUDE_DIR})
+                        PROPERTIES INTERFACE_INCLUDE_DIRECTORIES ${ARROW_INCLUDE_DIR}
+                                   IMPORTED_LOCATION ${ARROW_SHARED_LIB})
+  if(WIN32)
+    set_target_properties(${ARROW_LIBRARY_TARGET} PROPERTIES IMPORTED_IMPLIB
+                                                             ${ARROW_LINK_LIB})
```

Comment:
Can we always set `IMPORTED_IMPLIB` here?
I expect it's only used on Windows. (It's ignored on non Windows environments such as Linux and macOS.)

Reply:
Yes, it's true that the `IMPORTED_IMPLIB` variable is only used on Windows. Because imported libraries are not used on macOS and Linux, `CMAKE_IMPORT_LIBRARY_SUFFIX` and `CMAKE_IMPORT_LIBRARY_PREFIX` are not set on these platforms. They are used to create the file path contained by `ARROW_LINK_LIB`. Therefore, the value of `ARROW_LINK_LIB` is not set for macOS and Linux. 

Because of this, I would prefer to only set `IMPORTED_IMPLIB` on Windows.